lib/src/utf8.dart

```diff
@@ -39,13 +40,21 @@ extension Utf8Pointer on Pointer<Utf8> {
   /// If [length] is provided, zero-termination is ignored and the result can
   /// contain NUL characters.
   String toDartString({int? length}) {
+    _ensureNotNullptr();
     if (length != null) {
       RangeError.checkNotNegative(length, 'length');
     } else {
       length = this.length;
```

Comment:
This now does `_ensureNotNullPtr` twice, because `length` also does it.
Likely a small overhead, and if `length` *and* _`ensureNotNullPtr` are both inlined probably removable.
Consider having a non-checking `_length` helper that both can share.
Maybe:
```dart
static int _length(Pointer<Uint8> array) {
  var length = 0;
  while (array[length] != 0) {
    length++;
  }
  return length;
}
int get length {
  _ensureNotNullPtr("length");
  return _length(this);
}
```
and then use `_length(this)` here too. Can even avoid doing `cast<uint8>()` twice, if that has any cost.

Your call.